core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -66,6 +66,30 @@
 #include <sstream>
 #include <string>
 
+#ifdef KOKKOS_IMPL_DEBUG_CUDA_SERIAL_EXECUTION
```

Comment:
Is there any reason why people would want to build Kokkos without this feature?  (For example, does it require building kernels twice?)  If not, then I don't think it's necessary to have the feature macro.

Reply:
@mhoemmen , that's a really interesting approach I wouldn't have considered. @crtrott, do we have evidence that this impedes optimization or anything when turned on but unused?

Reply:
Yeah this will change code paths for scans quite a bit, and will increase compile time for those.

Reply:
Would it perhaps make sense for this to be enabled by default in debug builds?  It would be helpful to have regular testing.